### PR TITLE
Refactor: tests readability

### PR DESCRIPTION
**Pull request by lzukowski**

### CODE REVIEW

File: tests/events.py [hunk 1]
```diff
@@ -10,10 +7,3 @@ class SomeEvent(Event):
 
 class AnotherEvent(Event):
```
Comment (Enforcer) [inline tests/events.py:8]: Why did you left it here and moved other stuff out?

File: tests/event_store/bdd.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+@dataclass
+class When:
+    store: EventStore
+
+    def snapshotting(self, with_: Metadata, on: StreamId) -> Self:
+        self.store.save_snapshot(on, with_)
+        return self
+
+    def appending(self, *events: Metadata, on: StreamId) -> Self:
```
[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/bdd.py:65]: I was thinking about this. What reads better:
`when snapshotting`
or
`when snapshot`

I can change to a verb.

File: tests/event_store/test_save_retrieval.py [hunk 3]
```diff
@@ -3,82 +3,80 @@
...
-        events[2],
-        events[3],
-    ]
+def test_save_retrieve_part_of_stream(
+    given: Given,
+    when: When,
+    event_store: EventStore,
+) -> None:
+    given.stream(stream_id := StreamId())
+    when.appending(
```
[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/test_save_retrieval.py:24]: Ok. I will make it `given`, thx

File: tests/event_store/test_save_retrieval.py [hunk 4]
```diff
@@ -3,82 +3,80 @@
...
+        an_event := AnEvent(metadata={"correlation_id": uuid4(), "ip": "127.0.0.1"}),
+        on=stream_id,
     )
-    stream_id = StreamId(uuid4())
-
-    event_store.append(an_event, stream_id=stream_id)
-    events = event_store.load_stream(stream_id=stream_id)
-
-    assert events == [an_event]
+    then.stream(stream_id).events.equals_to(an_event)
```
[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/test_save_retrieval.py:48]: Yes, I agree.

File: tests/event_store/bdd.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/bdd.py:65]: :+1: thx

File: tests/event_store/test_snapshots.py [hunk 5]
```diff
@@ -1,54 +1,67 @@
...
-    pass
+def test_handles_snapshots(given: Given, when: When, then: Then) -> None:
+    given.stream(stream_id := StreamId())
+    given.events(AnEvent(), AnEvent(), AnEvent(), on=stream_id)
 
+    when.snapshotting(snapshot := Snapshot(), on=stream_id)
 
-class Snapshot(Event):
-    pass
+    then.stream(stream_id).events.equals_to(snapshot)
```
[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/test_snapshots.py:14]: Yes, this seems to be a better approach. Will fix it

File: tests/events.py (hunk 1 above)
Comment (lzukowski) [reply tests/events.py:8]: Cause there are still dependencies on other parts of tests. 

I will remove it after refactoring the rest.

File: tests/event_store/bdd.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/bdd.py]: Maybe
`when appends`, `when snapshots`, `when deletes` ?

[the remark this answers is not in the document]
Comment (Enforcer) [reply tests/event_store/bdd.py]: also sounds good

[the remark this answers is not in the document]
Comment (lzukowski) [reply tests/event_store/bdd.py]: So I would prefer this.

**Review by Enforcer — APPROVED** [review]

**Pull request merged**